x-pack/plugins/security/public/management/roles/edit_role/validate_role.test.ts

```diff
@@ -98,6 +98,85 @@ describe('validateRoleName', () => {
   });
 });
 
+describe('validateRoleName for serverless', () => {
+  beforeEach(() => {
+    validator = new RoleValidator({ shouldValidate: true, buildFlavor: 'serverless' });
+  });
+  test('should not allow whitespace', () => {
+    const role = {
+      name: 'role name',
+      elasticsearch: {
+        cluster: [],
+        indices: [],
+        run_as: [],
+      },
+      kibana: [],
+    };
+    expect(validator.validateRoleName(role)).toEqual({
+      isInvalid: true,
+      error: `Name must contain only alphanumeric characters, and non-leading dots, hyphens, or underscores.`,
+    });
+  });
+  test('should not allow leading symbols', () => {
+    const role = {
+      name: '.rolename',
+      elasticsearch: {
+        cluster: [],
+        indices: [],
+        run_as: [],
+      },
+      kibana: [],
+    };
+    expect(validator.validateRoleName(role)).toEqual({
+      isInvalid: true,
+      error: `Name must contain only alphanumeric characters, and non-leading dots, hyphens, or underscores.`,
+    });
+  });
+  test('should allow underscores contained', () => {
+    const role = {
+      name: 'role_name',
+      elasticsearch: {
+        cluster: [],
+        indices: [],
+        run_as: [],
+      },
+      kibana: [],
+    };
+    expect(validator.validateRoleName(role)).toEqual({
+      isInvalid: false,
+    });
+  });
+  test('should allow valid names', () => {
+    const role = {
+      name: 'rolename_',
+      elasticsearch: {
+        cluster: [],
+        indices: [],
+        run_as: [],
+      },
+      kibana: [],
+    };
+    expect(validator.validateRoleName(role)).toEqual({
+      isInvalid: false,
+    });
+  });
+  test('should not allow any special characters except for underscore, dots and hyphens', () => {
```

Comment:
Nit: same as above - for full coverage, I would expect to see each of the unallowed characters tested. Similar to the test pattern used in the "it allows the "${element}" character" test, except validating a negative result.